pkg/schemas/service.go

```diff
@@ -223,6 +227,59 @@ func (s *Service) DeleteSchema(ctx context.Context, ID string) error {
 
 }
 
+func (s *Service) GetDefaultSchema(ctx context.Context) (*DefaultSchema, error) {
+	ctx, span := s.tracer.Start(ctx, "schemas.Service.GetDefaultSchema")
+	defer span.End()
+
+	cm, err := s.k8s.ConfigMaps(s.cmNamespace).Get(ctx, s.cmName, metaV1.GetOptions{})
+
+	if err != nil {
+		s.logger.Error(err.Error())
+		return nil, err
+	}
+
+	ID, ok := cm.Data[DEFAULT_SCHEMA]
+
+	if !ok {
+		return nil, fmt.Errorf("default schema %s missing", DEFAULT_SCHEMA)
+	}
+
+	defaultSchema := new(DefaultSchema)
+	defaultSchema.ID = ID
+
+	return defaultSchema, nil
+}
+
+func (s *Service) UpdateDefaultSchema(ctx context.Context, schema *DefaultSchema) (*DefaultSchema, error) {
+	ctx, span := s.tracer.Start(ctx, "schemas.Service.UpdateDefaultSchema")
+	defer span.End()
+
+	cm, err := s.k8s.ConfigMaps(s.cmNamespace).Get(ctx, s.cmName, metaV1.GetOptions{})
+
+	if err != nil {
+		s.logger.Error(err.Error())
+		return nil, err
+	}
+
+	if _, ok := cm.Data[DEFAULT_SCHEMA]; !ok {
+		return nil, fmt.Errorf("default schema %s missing", DEFAULT_SCHEMA)
+	}
+
+	if schema.ID == DEFAULT_SCHEMA {
+		return nil, fmt.Errorf("schema with ID %s not available", schema.ID)
+	} else if _, ok := cm.Data[schema.ID]; !ok && schema.ID != cm.Data[DEFAULT_SCHEMA] {
```

Comment:
If the user provides the same schema ID, why would we throw an error? The second check should be removed.

I think that the logic here should be just that the `schema.ID` provided is in the map and that it does not have the `DEFAULT_SCHEMA` value.

Reply:
This was misguided. I fixed it.